src/main/kotlin/com/zepben/evolve/database/sqlite/metrics/MetricsEntryWriter.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2024 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.database.sqlite.metrics
+
+import com.zepben.evolve.database.sqlite.common.BaseEntryWriter
+import com.zepben.evolve.database.sqlite.extensions.setInstant
+import com.zepben.evolve.database.sqlite.metrics.tables.TableJobSources
+import com.zepben.evolve.database.sqlite.metrics.tables.TableJobs
+import com.zepben.evolve.database.sqlite.metrics.tables.TableNetworkContainerMetrics
+import com.zepben.evolve.metrics.*
+import java.util.*
+
+/**
+ * Class for writing entries to the metrics database.
+ *
+ * @param databaseTables The tables in the metrics database.
+ * @param jobId The ID of the job to write entries for.
+ */
+class MetricsEntryWriter(
+    private val databaseTables: MetricsDatabaseTables,
+    private val jobId: UUID
+) : BaseEntryWriter() {
+
+    /**
+     * Save an [IngestionMetadata] to the `jobs` table.
+     *
+     * @param metadata the [IngestionMetadata] to save.
+     * @return true if the [metadata] saved successfully or [metadata] is null.
+     */
+    fun save(metadata: IngestionMetadata?): Boolean {
+        if (metadata == null) return true // signals "saved 0 metadata successfully"
```

Comment:
Should this instead return false, preventing other tables from saving if there is no metadata for a job?

Reply:
Otherwise, we may end up with job sources/metrics for jobs not present in the `jobs` table.

Reply:
yeah we should avoid having things that don't have references back to jobs - so happy to prevent it.

Reply:
Added check in `MetricsWriter`.